aiidalab_widgets_base/process.py

```diff
@@ -30,13 +30,28 @@ def get_running_calcs(process):
 class SubmitButtonWidget(VBox):
     """Submit button class that creates submit button jupyter widget."""
 
-    def __init__(self, process, input_dictionary_function, description="Submit"):
-        """Submit Button
-        :process: work chain to run
-        :param_funtion: the function that generates input parameters dictionary
+    def __init__(self,
+                 process,
+                 input_dictionary_function,
+                 description="Submit",
+                 disable_after_submit=True,
+                 append_output=False):
+        """Submit Button widget.
+
+        process (Process): Process to submit.
+
+        input_dictionary_function (func): Function that generates input parameters dictionary.
+
+        description (str): Description written on the submission button.
+
+        disable_after_submit (bool): Whether to disable the button after the process was submitted.
+
+        append_output (bool): Whether to clean the widget output before every submission.
         """
 
         self.process = None
```

Comment:
Since it is now possible to submit multiple times. What is the expected behavior here? We override this attribute and lose track of previous processes?

Reply:
Yes, I think it would even make sense for this guy to become a trait. This way the change can be detected and the process can be stored elsewhere.

What do you think?

Reply:
Made it in c301acdee3fc85686943f366e49f2d8ee0fb0d95